overlord/state/taskrunner.go

```diff
@@ -94,6 +102,7 @@ func (r *TaskRunner) run(t *Task) {
 		panic("internal error: attempted to run task with nil handler for status " + t.Status().String())
 	}
 
+	t.scheduledTime = time.Time{} // clear schedule
```

Comment:
I believe we have a clear separation between the task runner and the task internals for now. It'd be nice to keep it that way, if we can.

The above may be `time.At(time.Time{})`, except it's probably the wrong place. We should reset it after we've run it, not before. Otherwise a crash over the next few lines will prematurely clean the schedule.